mldsa/poly.c

```diff
@@ -304,10 +304,17 @@ void mld_poly_use_hint(mld_poly *b, const mld_poly *a, const mld_poly *h)
  * that it is okay to leak which coefficient violates the bound (while the
  * coefficient itself must remain secret).
  * We instead perform everything in constant-time.
+ * Also it is sufficient to check that it is smaller than
+ * MLDSA_Q - REDUCE32_RANGE_MAX > (MLDSA_Q - 1) / 8).
  */
 MLD_INTERNAL_API
 uint32_t mld_poly_chknorm(const mld_poly *a, int32_t B)
 {
+#if defined(MLD_USE_NATIVE_POLY_CHKNORM)
+  /* TODO: proof */
```

Comment:
Do we need to add a proof dir for "poly_chknorm_native" below proofs/cbmc to exercise this?

Reply:
Yes, we have an issue for that: https://github.com/pq-code-package/mldsa-native/issues/350

The proofs itself are straightforward, but what needs to happen before that is that for each function we need to add comments to the asm documenting the bound propagation and derive the suitable contract that matches alll backends. For poly_chknorm this is trivial, but for the NTT needs a bit of thinking.